### PR TITLE
Support match for Teaser

### PR DESCRIPTION
**Pull request by chezzdev**
Checks:

* [ ] Update changelog
* [x] Rebase to `dev` branch
* [x] Assign reviewers

Linked issues:
Resolves #2134

### CODE REVIEW

File: Matchfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# The docs are available on https://docs.fastlane.tools/actions/match
+
+git_url("https://github.com/mapbox/apple-certificates")
+storage_mode("git")
+
+app_identifier("com.mapbox.visionsdk.demo")
+
+# Possible values appstore, adhoc, enterprise or development
```
Comment (dersim-davaod) [inline Matchfile]: nit:

```suggestion
# Possible values appstore, adhoc, enterprise, development or developer_id
```

?

File: Matchfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+type("appstore")
+readonly(true)
```
Comment (dersim-davaod) [inline Matchfile]: Should we add
```
shallow_clone(true)
verbose(true)
```

?

File: Matchfile (hunk 1 above)
Comment (chezzdev) [reply Matchfile]: It doesn't seem that other types except for `development` and `appstore` are applicable in this case. I think I'll leave just those two. @dersim-davaod agree?

File: Matchfile (hunk 2 above)
Comment (chezzdev) [reply Matchfile]: Ok with `shallow_clone`, `verbose` is definitely redundant in general case.

File: Matchfile (hunk 1 above)
Comment (dersim-davaod) [reply Matchfile]: Absolutely :)

**Pull request merged**